src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -1663,17 +1665,17 @@ public void Joining(UniqueAddress node, ImmutableHashSet<string> roles, AppVersi
         /// <exception cref="InvalidOperationException">Welcome can only be done from an empty state</exception>
         public void Welcome(Address joinWith, UniqueAddress from, Gossip gossip)
         {
-            if (!_latestGossip.Members.IsEmpty) throw new InvalidOperationException("Welcome can only be done from an empty state");
+            if (!LatestGossip.Members.IsEmpty) throw new InvalidOperationException("Welcome can only be done from an empty state");
             if (!joinWith.Equals(from.Address))
             {
                 _cluster.LogInfo("Ignoring welcome from [{0}] when trying to join with [{1}]", from.Address, joinWith);
             }
             else
             {
                 _cluster.LogInfo("Welcome from [{0}]", from.Address);
-                _latestGossip = gossip.Seen(SelfUniqueAddress);
+                _membershipState = _membershipState.Copy(gossip).Seen();
```

Comment:
Ah, this is probably why our tests were failing - forgot to copy the initial `Gossip` payload sent by the leader during a `Welcome` operation